articles/113_ros_time.md

```diff
@@ -0,0 +1,115 @@
+
+The final challenge is that to be useful for log file playback the time abstraction must be able to jump backwards in time.
+This is a very different behavior from most clocks, and requires developers using the time abstraction to make sure their algorithm can deal with the discontinuity.
+
+## System Behavior
+
+To implement the time abstraction the following approach will be used.
+
+The time abstraction can be published by one source on the `/clock` topic.
+The topic will contain the latest abstracted time for the ROS system.
+If a publisher exists for the topic it will override the system time when using the ROS time abstraction.
+If `/clock` is being published calls to the ROS time abstraction will return the latest time received from the `/clock` topic with a default of zero if nothing has been received.
+
+If the time on the clock jumps backwards a callback handler will be invoked and be required to complete before any calls to the ROS time abstraction report the new time.
+Calls that come in before that must block.
+The developer has the opportunity to register callbacks with the handler to clear any state from their system if necessary before time will be in the past.
+
+The frequency of publishing the `/clock` as well as the granularity are not specified as they are application specific.
+
+### No Advanced Estimating Clock
+
+There are more advanced techniques which could be included to attempt to estimate the propagation properties and extrapolate between time ticks.
+However all of these techniques will require making assumptions about the future behavior of the time abstraction.
+And in the case that playback or simulation is instantaneously paused it will break any of these assumptions.
+There are techniques which would allow potential interpolation, however to make these possible it would require providing guarantees about the continuity of time into the future.
+For more accuracy the progress of time can be slowed, or the frequency of publishing can be increased.
+These are both 
+
+## Public API
+
+In each client library there will be provided an abstraction of `Time`, `Duration`, and `Rate`.
+The duration abstraction will support a `sleep` function which 
+It will also provide a `Timer` object which will provide periodic callback functionality using the ROS time abstraction.
+
+### Walltime
+
+There are several use cases where using the ROS time abstraction are inappropriate.
+Example use cases for this include hardware drivers which are interacting with peripherals with hardware timeouts.
+For convenience in these cases we will also provide the same API as above, but use the name `WallTime`.
+
+`WallTime` will be directly tied to the system clock.
+
+In nodes which require the use of `WallTime` for interacting with hardware or other peripherals it is expected that they do a best effort to isolate any `WallTime` information inside their implementation and translate external interfaces to use the ros time abstraction.
```

Comment:
I'm going to update it to use 
system, steady, and ros, instead of the old rostime and walltime.